src/main/java/io/divolte/server/DivolteSchema.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2015 GoDataDriven B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.divolte.server;
+
+import org.apache.avro.Schema;
+
+import javax.annotation.ParametersAreNonnullByDefault;
+import java.util.Optional;
+
+@ParametersAreNonnullByDefault
+public class DivolteSchema {
```

Comment:
since this is the "schema" in the Divolte schema registry, it is interpreted by the different sink types.  As the same mapping can refer to multiple sinks, polymorphism could lead to a combinatorial explosion.

Perhaps a better way would just be composition - enrich this class with getConfluentMapping, getDivolteMapping, etc. methods.  Those can return specialized configuration objects.